src/components/Button/Button.stories.tsx

```diff
@@ -1,15 +1,18 @@
 import type { Meta, StoryObj } from "@storybook/react";
 import { fn } from "@storybook/test";
 
-import { InkIcon } from "../..";
-import { AllButtonVariants, AllButtonVariantsProps } from "./AllButtonVariants";
+import { Button, ButtonProps, InkIcon } from "../..";
+import { MatrixDecorator } from "../../decorators/MatrixDecorator";
 
-const meta: Meta<AllButtonVariantsProps> = {
-  title: "Example/Button",
-  component: AllButtonVariants,
-  parameters: {
-    layout: "centered",
-  },
+const meta: Meta<ButtonProps<"button" | "a">> = {
+  title: "Components/Button",
+  decorators: [
+    MatrixDecorator<ButtonProps>({
```

Comment:
I figured I could create the same behaviour as the custom "AllButtonVariants" as a decorator.

This is very useful because Props are better represented if we use the actual component instead of a wrapper

Reply:
That's pretty cool! 🙂 Right now it only supports two props, I guess we could set it up to support any amount of props? TBD later.

Reply:
Yeah but complexity went higher if I had made it super generic, so I figured why no just 2 for now ;P